[develop] Enable UPP 2d decomposition

MichaelLueken opened the pull request:
## DESCRIPTION OF CHANGES: 
With recent changes made to the UFS-WM (PR [#1823](https://github.com/ufs-community/ufs-weather-model/pull/1823)), it is now possible to see in the log files that UPP 2d decomposition is being performed for inline post.  To this end, the UFS-WM hash has been updated to [020e783](https://github.com/ufs-community/ufs-weather-model/commit/020e783a130c500f628cc221853afe523e5728bf) (October 27, 2023), the UPP hash was updated to [fae617b](https://github.com/NOAA-EMC/UPP/commit/fae617ba485dbbadc8fc10f512a6a0c29c81741a) (October 6, 2023), and the UFS_UTILS hash was updated to [dc0e4a6](https://github.com/ufs-community/UFS_UTILS/commit/dc0e4a61419b95805a22e05d477cb58a33e581f8) (November 6, 2023).

With the updated UPP hash, the old `postxconfig-NT-fv3lam.txt` file had been removed.  Wen suggested using the `postxconfig-NT-fv3lam_rrfs.txt` file instead.  Unfortunately, due to issues with the CRTM on Derecho, we are unable to move forward with this postxconfig file (this file contains assimilated radiances).  The ufs-weather-model/tests/parm directory contains a copy of the old `postxconfig-NT-fv3lam.txt` file.  To expedite moving forward with this update, I'm currently pointing to this copy of the postxconfig file.  Once the issue with the CRTM on Derecho has been corrected, the SRW App repository can transition to the `postxconfig-NT-fv3lam_rrfs.txt` file.

Changes to enable 2d decomposition include:
- parm/model_configure - Added `itasks` to the `model_configure` file (values greater than 1 enable 2d decomposition in inline post).
- scripts/exregional_run_post.sh - Added `numx` to the end of the `&NAMPGB` namelist options (values of numx greater than 1 enable 2d decomposition in offline post).
- ush/create_model_configure_file.py - Added `itasks` to the list of variables to be added to the `model_configure` file.
- tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2.yaml - Added `ITASKS: 2` to enable inline post 2d decomposition.
- tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta.yaml - Added `NUMX: 2` to enable offline post 2d decomposition.

The ability to run comprehensive tests has been added back into the Jenkinsfile, as well as the ability to run the automated Jenkins tests on Derecho.

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 
- [X] hera.intel
- [X] hercules.intel
- [X] orion.intel
- [X] derecho.intel
- [X] gaea.intel
- [X] gaea-c5.intel
- [X] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [X] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)
The 2D decomposition does not work with GNU compilers (the `run_post` and `run_fcst` tasks fail on Hera GNU with double free or corruption errors).  Disabling 2D decomposition for GNU compilers will allow the updated 2D decomposition WE2E tests to run on Hera GNU.

## DEPENDENCIES:
None

## DOCUMENTATION:
Documentation has been updated in the WE2E test configuration files, the `ConfigWorkflow.rst` chapter, and `config_defaults.yaml`.

## ISSUE: 
Resolves #480

## CHECKLIST
- [X] My code follows the style guidelines in the Contributor's Guide
- [X] I have performed a self-review of my own code using the Code Reviewer's Guide
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

MichaelLueken committed:
[develop] Update UFS-WM and UPP hashes to enable inline/offline post 2d decomposition.

MichaelLueken committed:
[develop] Update run_fcst, run_post, and plot_allvars scripts to use CRTM coefficients and add CRTM_DIR to Hera machine yaml file.

MichaelLueken committed:
[develop] Enable post 2d decomposition for both inline and offline post.  Add itasks: 2 to the model_configure file for the grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2 test for inline post 2d decomposition and add numx: 2 to the itag namelist option file for the grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta test for offline post 2d decomposition.

MichaelLueken committed:
[develop] Update ush/machine/jet.yaml to include path to CRTM coefficient directory.

MichaelLueken committed:
[develop] Update ush/machine/gaea.yaml to include path to CRTM coefficient directory.

MichaelLueken committed:
Merge branch 'ufs-community:develop' into feature/upp_2d_decomp

MichaelLueken committed:
[develop] Update ush/machine/cheyenne.yaml to include path to CRTM coefficient directory.

MichaelLueken committed:
[develop] Update ush/machine/derecho.yaml to include path to CRTM coefficient directory and cleaned up comment in build_derecho_intel.lua.

MichaelLueken committed:
[develop] Update ush/machine/orion.yaml to include path to CRTM coefficient directory.

MichaelLueken committed:
[develop] Updates to allow the test_create_model_configure_file unit test to pass.

MichaelLueken committed:
[develop] Updates to allow the linting and test_create_model_configure_file unit tests to pass.

MichaelLueken committed:
[develop] Update to allow the test_generate_FV3LAM_wflow unit test to pass.

MichaelLueken committed:
Merge branch 'ufs-community:develop' into feature/upp_2d_decomp

MichaelLueken committed:
[develop] Update ush/machine/hercules.yaml to include path to CRTM coefficient directory.

MichaelLueken committed:
[develop] Update ush/machine/gaea-c5.yaml to include path to CRTM coefficient directory.

MichaelLueken committed:
Merge remote-tracking branch 'upstream/develop' into feature/upp_2d_decomp

MichaelLueken committed:
Merge branch 'develop' into feature/upp_2d_decomp

MichaelLueken committed:
[develop] Update ufs-weather-model, UPP, and UFS_UTILS hashes to more up-to-date versions.  Use configxpost-NT-fv3lam.txt from ufs-weather-model/tests/parm instead of configxpost-NT-fv3lam_rrfs.txt until CRTM issue is addressed on Derecho.  Add Derecho label to Jenkinsfile and reactivate comprehensive testing in Jenkins.

MichaelLueken committed:
[develop] Set USE_CRTM to false in config_defaults.yaml.

MichaelLueken committed:
[develop] Disable 2D decomposition for GNU.

MichaelLueken committed:
[develop] Update ConfigWorkflow.rst to add changes to ush/config_defaults.yaml.

MichaelLueken committed:
[develop] Correct ConfigWorkflow.rst default documentation for CUSTOM_POST_CONFIG_FP.

gspetro-NOAA commented on docs/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst:
```diff
@@ -1138,6 +1138,9 @@ These parameters set values in the Weather Model's ``model_configure`` file.
    Flag that determines whether to use the inline post option, which calls the Unified Post Processor (:term:`UPP`) from within the UFS Weather Model. The default ``WRITE_DOPOST: false`` does not use the inline post functionality, and the ``run_post`` tasks are called from outside of the UFS Weather Model. If ``WRITE_DOPOST: true``, the ``WRITE_DOPOST`` flag in the ``model_configure`` file will b ...
 
+``ITASKS``: (Default: 1)
+   Variable denoting the number of write tasks in the i direction in the current group.  Used for inline post 2d decomposition.  Setting this variable to a value greater than 1 will enable 2d decomposition.
```
```suggestion
   Variable denoting the number of write tasks in the ``i`` direction in the current group. Used for inline post 2D decomposition. Setting this variable to a value greater than 1 will enable 2D decomposition.
```

gspetro-NOAA commented on docs/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst:
```diff
@@ -1277,10 +1280,10 @@ Customized Post Configuration Parameters
    Flag that determines whether a user-provided custom configuration file should be used for post-processing the model data. If this is set to true, then the workflow will use the custom post-processing (:term:`UPP`) configuration file specified in ``CUSTOM_POST_CONFIG_FP``. Otherwise, a default configuration file provided in the UPP repository will be used. Valid values: ``True`` | ``False``
 
-``CUSTOM_POST_CONFIG_FP``: (Default: "")
+``CUSTOM_POST_CONFIG_FP``: (Default: ``"{{ user.SORCdir }}/ufs-weather-model/tests/parm/postxconfig-NT-fv3lam.txt"``)
```
@MichaelLueken Have you created an issue to change this back once we can point to `postxconfig-NT-fv3lam_rrfs.txt`? Just don't want to forget about it!

MichaelLueken replied:
I have an open issue, #916, which talks about the issues with the CRTM on Derecho.  I have gone in and added the temporary fix and that the UPP's `postxconfig-NT-fv3lam_rrfs.txt` file will be used once the CRTM is functioning properly on that machine.

gspetro-NOAA commented on docs/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst:
```diff
@@ -1295,6 +1298,9 @@ Set parameters for customizing the :term:`UPP`.
    The file that lists grib2 fields to be extracted for testbed files. An empty string means no need to generate testbed files.
 
+``NUMX``: (Default: 1)
+   The number of i regions in a 2D decomposition.  Each i row is distibuted to numx ranks.  Set NUMX to a value greater than 1 to enable 2D decomposition.
```
```suggestion
   The number of ``i`` regions in a 2D decomposition. Each ``i`` row is distributed to ``NUMX`` ranks. Set ``NUMX`` to a value greater than 1 to enable 2D decomposition.
```

gspetro-NOAA commented on tests/WE2E/run_WE2E_tests.py:
```diff
@@ -228,6 +228,13 @@ def run_we2e_tests(homedir, args) -> None:
+                test_cfg['task_run_post'].update({"NUMX": 1})
+            if 'task_run_fcst' in test_cfg:
+                test_cfg['task_run_fcst'].update({"ITASKS": 1})
+
```
For this section, it might be good to also print some kind of warning message so that users know that this is happening.

MichaelLueken replied:
Great idea!  I have added logging information that NUMX and ITASKS are being reset for GNU compilers.

MichaelLueken replied:
I have replaced logging.debug with logging.info.  Resolving conversation now.

gspetro-NOAA commented on tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2.yaml:
```diff
@@ -3,7 +3,8 @@ metadata:
     completes successfully on the RRFS_CONUS_25km grid using the GFS_v15p2
     physics suite with ICs and LBCs derived from the FV3GFS.
-    In addition, this tests the "inline post" option (WRITE_DOPOST: true)
+    In addition, this tests the "inline post" option (WRITE_DOPOST: true).
+    Enables UPP inline 2d decomposition.
```
```suggestion
    In addition, this tests the "inline post" option (WRITE_DOPOST: true)
    and enables UPP inline 2D decomposition.
```

gspetro-NOAA commented on tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta.yaml:
```diff
@@ -3,7 +3,7 @@ metadata:
     completes successfully on the RRFS_CONUScompact_25km grid using the RRFS_v1beta
     physics suite with ICs derived from the HRRR and LBCs derived from the RAP.
-    Also tests the "DOT_OR_USCORE" option
+    Also tests the "DOT_OR_USCORE" option.  Enables offline UPP 2d decomposition.
```
```suggestion
    It also tests the "DOT_OR_USCORE" option and enables offline UPP 2d decomposition.
```

gspetro-NOAA commented on docs/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst:
```diff
@@ -1295,6 +1298,9 @@ Set parameters for customizing the :term:`UPP`.
 
+``NUMX``: (Default: 1)
+   The number of i regions in a 2D decomposition.  Each i row is distibuted to numx ranks.  Set NUMX to a value greater than 1 to enable 2D decomposition.
+
```
```suggestion
   Note that 2D decomposition does not yet work with GNU compilers, so this value will be reset to 1 automatically when using GNU compilers (i.e., when ``COMPILER: gnu``).

```

gspetro-NOAA commented on docs/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst:
```diff
@@ -1138,6 +1138,9 @@ These parameters set values in the Weather Model's ``model_configure`` file.
 
+``ITASKS``: (Default: 1)
+   Variable denoting the number of write tasks in the i direction in the current group.  Used for inline post 2d decomposition.  Setting this variable to a value greater than 1 will enable 2d decomposition.
+
```
```suggestion
   Note that 2D decomposition does not yet work with GNU compilers, so this value will be reset to 1 automatically when using GNU compilers (i.e., when ``COMPILER: gnu``).

```

gspetro-NOAA commented on ush/config_defaults.yaml:
```diff
@@ -1787,12 +1787,19 @@ task_run_fcst:
+  # current group.  Used for inline post 2d decomposition.  Setting this
+  # variable to a value greater than 1 will enable 2d decomposition.
+  # Default setting is 1.
+  #
```
```suggestion
  # Note that 2D decomposition does not yet work with GNU compilers, so this value 
  # will be reset to 1 automatically when using GNU compilers (i.e., when COMPILER: gnu).
  #
```

MichaelLueken replied:
I've also replaced the lowercase d with uppercase D in the 2D decomposition documentation for ITASKS above this line.

gspetro-NOAA commented on ush/config_defaults.yaml:
```diff
@@ -2067,12 +2074,17 @@ task_run_post:
+  # NUMX:
+  # The number of i regions in a 2D decomposition.  Each i row is 
+  # distibuted to numx ranks. Default value of numx is 1.
+  #
```
```suggestion
  # Note that 2D decomposition does not yet work with GNU compilers, so this value 
  # will be reset to 1 automatically when using GNU compilers (i.e., when COMPILER: gnu).
  #
```

MichaelLueken committed:
[develop] Update docs/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst with Gillian's suggestions.

Co-authored-by: Gillian Petro <96886803+gspetro-NOAA@users.noreply.github.com>

MichaelLueken committed:
[develop] Update docs/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst with Gillian's suggestions.

Co-authored-by: Gillian Petro <96886803+gspetro-NOAA@users.noreply.github.com>

MichaelLueken committed:
[develop] Update tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2.yaml with Gillian's suggestions.

Co-authored-by: Gillian Petro <96886803+gspetro-NOAA@users.noreply.github.com>

MichaelLueken committed:
[develop] Update tests/WE2E/test_configs/grids_extrn_mdls_suites_community/config.grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta.yaml with Gillian's suggestions.

Co-authored-by: Gillian Petro <96886803+gspetro-NOAA@users.noreply.github.com>

MichaelLueken committed:
[develop] Update docs/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst with Gillian's suggestions.

Co-authored-by: Gillian Petro <96886803+gspetro-NOAA@users.noreply.github.com>

MichaelLueken committed:
[develop] Update docs/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst with Gillian's suggestions.

Co-authored-by: Gillian Petro <96886803+gspetro-NOAA@users.noreply.github.com>

MichaelLueken committed:
[develop] Update ush/config_defaults.yaml with Gillian's suggestions.

Co-authored-by: Gillian Petro <96886803+gspetro-NOAA@users.noreply.github.com>

MichaelLueken committed:
[develop] Replace lowercase d with uppercase D for ITASKS 2D decomposition documentation in ush/config_defaults.yaml and added prints regarding resetting NUMX and ITASKS in tests/WE2E/run_WE2E_tests.py.

MichaelLueken commented:
@gspetro-NOAA - Thank you very much for reviewing these changes!  I have committed the suggestions that you have made to the documentation, replaced the lowercase d with uppercase D for ITASKS in `config_defaults.yaml` for consistency, and added logging prints to `run_WE2E_tests.py` to let users know that the values are being reset for GNU compilers.  I'm hoping I'll be able to satisfactorily go through this morning's code review!  Please let me know if you see any other changes that should be made or if you have any questions.

MichaelLueken committed:
[develop] Replace logging.debug with logging.info for NUMX and ITASKS reset in run_WE2E_tests.py and replace lowercase d with uppercase D in grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta documentation.

gspetro-NOAA commented:
It looks like we need to add the line:
```
DOMAIN_PREGEN_BASEDIR: /contrib/EPIC/UFS_SRW_data/develop/FV3LAM_pregen
```
to the noaacloud machine file for the fundamental tests to run on the cloud. @EdwardSnyder-NOAA said that we did this for the release in PR #937 and should do that in `develop`, too. Without it, experiment generation failed. After I added that line, fundamental tests generally pass.
- On AWS, only `grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta` fails, but it is failing due to hitting the walltime for the post-processing tasks. It passes on GCP.
- On GCP, only `grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_RAP_suite_WoFS_v0` fails due to hitting the walltime for the run_fcst task. It passes on AWS. 
I can approve once the machine file change is made.

MichaelLueken committed:
[develop] Add DOMAIN_PREGEN_BASEDIR entry to ush/machine/noaacloud.yaml to allow for workflow generation on NOAA Cloud platforms.

MichaelLueken commented:
Thank you very much for running the fundamental tests on NOAA Cloud platforms, @gspetro-NOAA!  I have added the `DOMAIN_PREGEN_BASEDIR` entry into `ush/machine/noaacloud.yaml` at 
[ec44d97](https://github.com/ufs-community/ufs-srweather-app/pull/917/commits/ec44d97b437fd3011211190bc9d74caaca837c65).

gspetro-NOAA reviewed (approved):
Looks good to me! Approving...

MichaelLueken committed:
Merge branch 'ufs-community:develop' into feature/upp_2d_decomp

MichaelLueken committed:
Merge branch 'ufs-community:develop' into feature/upp_2d_decomp

MichaelLueken committed:
Merge branch 'ufs-community:develop' into feature/upp_2d_decomp

mkavulich commented on docs/UsersGuide/source/CustomizingTheWorkflow/ConfigWorkflow.rst:
```diff
@@ -1295,6 +1299,10 @@ Set parameters for customizing the :term:`UPP`.
 ``TESTBED_FIELDS_FN``: (Default: "")
    The file that lists grib2 fields to be extracted for testbed files. An empty string means no need to generate testbed files.
 
+``NUMX``: (Default: 1)
```
I'm a bit confused from this description. Is ``i`` here the same as `ITASKS`? If not, can you expand a little bit more on the relationship between `ITASKS` and `NUMX`?

MichaelLueken replied:
@mkavulich For offline post, there is no `ITASKS`.  `ITASKS` is only used in inline post and is set in `ufs-weather-model/FV3/io/post_fv3.F90` as:

`numx = itasks`

UPP requires a value of `NUMX` greater than 1 to activate 2d decomposition, and rather than add `NUMX` to inline post, the `ITASKS` variable was used instead.

So, in the case of offline post, no, `i` is not the same as `ITASKS` (since there is no `ITASKS` variable in UPP).  For inline post, `i` is the same as `ITASKS`.

mkavulich replied:
Sorry, I'm still a bit confused: so inline and offline post have two different ways of enabling 2d-decomposition? NUMX, used for inline post only, is treated the same as ITASKS is in offline post?

MichaelLueken replied:
Documentation for `NUMX` has been updated to show that it is for offline post (while `ITASKS` is for inline post).

mkavulich reviewed (commented):
Just one question before I approve

MichaelLueken committed:
Merge remote-tracking branch 'upstream/develop' into feature/upp_2d_decomp

mkavulich reviewed (approved):
Sorry I let this slide off my plate: I'll approve now so I don't hold up this PR any longer: my only concern was a bit of confusion on two different parameters controlling the same thing (one for inline post, one for offline post). But this is a documentation issue if it is an issue at all.

MichaelLueken committed:
Merge remote-tracking branch 'upstream/develop' into feature/upp_2d_decomp

MichaelLueken committed:
[feature/upp_2d_decomp] Update numx documentation to note that it is used for offline post 2D decomposition

MichaelLueken commented:
On Jet, the `grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2` WE2E coverage test had failed.  Rocotorewind/rocotoboot allowed the test to successfully pass:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
community_20240216161033                                           COMPLETE              20.37
custom_ESGgrid_20240216161035                                      COMPLETE              17.82
custom_ESGgrid_Great_Lakes_snow_8km_20240216161036                 COMPLETE              58.08
custom_GFDLgrid_20240216161038                                     COMPLETE               9.41
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_nemsio_2021032018_202402  COMPLETE              10.73
get_from_HPSS_ics_FV3GFS_lbcs_FV3GFS_fmt_netcdf_2022060112_48h_20  COMPLETE              55.55
get_from_HPSS_ics_RAP_lbcs_RAP_20240216161043                      COMPLETE              17.62
grid_RRFS_AK_3km_ics_FV3GFS_lbcs_FV3GFS_suite_HRRR_20240216161044  COMPLETE             338.41
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              45.93
grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v15p2_20240  COMPLETE               8.29
grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_RRFS_v1beta_2024  COMPLETE             523.51
nco_grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR_2024  COMPLETE              10.59
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE            1116.31
```
This also shows that the code in this branch is ready to run on Rocky8, since the xjet partition is required to run the forecast job.  One caveat is that the Jenkins pipeline for Jet will need to be set to one of the login nodes that has been transitioned to Rocky8, or else the `run_fcst` job will never run (you need to be logged into a Rocky8 login node in order to use the xjet partition).

Once a successful run of the `nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_GFS_v16` WE2E test completes on Derecho, this work can get merged.  Currently encountering issues with the `get_extrn_ics` and `get_extrn_lbcs` tasks failing to retrieve the necessary data from AWS in 3 hours (this has only been encountered beginning after the Presidents' Day weekend).  If this task continues to fail, then the memory for these tasks will need to be raised from 2G to a higher value.

MichaelLueken commented:
Manually submitting the Jenkins coverage WE2E tests on Derecho successfully passed:
```
----------------------------------------------------------------------------------------------------
Experiment name                                                  | Status    | Core hours used 
----------------------------------------------------------------------------------------------------
custom_ESGgrid_IndianOcean_6km_20240221070627                      COMPLETE              30.98
grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16_plot_20  COMPLETE              49.38
grid_RRFS_CONUS_25km_ics_NAM_lbcs_NAM_suite_GFS_v16_2024022107063  COMPLETE              54.23
grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_HRRR_20240221  COMPLETE              36.72
grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta_2  COMPLETE              22.19
grid_SUBCONUS_Ind_3km_ics_HRRR_lbcs_HRRR_suite_HRRR_2024022107064  COMPLETE              48.32
nco_grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_timeoffset_suite_  COMPLETE              30.16
pregen_grid_orog_sfc_climo_20240221070649                          COMPLETE              23.06
specify_template_filenames_20240221070651                          COMPLETE              23.59
----------------------------------------------------------------------------------------------------
Total                                                              COMPLETE             318.63
```
Moving forward with merging this work now.
